### ISSUE TITLE
Ordering of Configuration and Startup Classes intellisense ML

### ISSUE THREAD

**Issue by davidbuckleyni** [dotnet/aspnetcore#36192]
### Is your feature request related to a problem? Please describe.
Even as seasoned developer often at times we can get lost in code and get confused on the ordering of startup configs and app using.

### Describe the solution you'd like
I would think a way that ML Intellense could be updated to show some kind of ordering system to certain use cases for example If the user adds in EF Core Authentication 

It would alert the user that they require app.UsetAuthhorization() but it should be at a specfic order in the config section

### Additional context
For Example here 

            app.UseNToastNotify();

            app.UseSession();
            app.UseHttpsRedirection();
            app.UseStaticFiles();
            app.UseRouting();
            app.UseAuthorization();  ->> wrong order
            app.UseCustomizedStaticFiles(env);

            app.UseAuthentication();
            app.UseAuthorization();-->correct order

In much the same way that when creating auto properties using tab the intellisense would suggest the next action if something was installed like UseAuthorization is it possible that the ML and Interlines teams could look at some kind of ordering suggestions, would help programmers not loose hours over some misplaced configuration steps.

For Example we could have some kind of of placement arrow come up   that would be to the left of the app. that would suggest the first instance is in the wrong order?

**Comment by Pilchie** [comment]
Typically we do these with analyzers.

**Comment by davidbuckleyni** [comment]
> Typically we do these with analyzers.

Agreed but for the beginner it can be frustrating while im a senior programmer beginners have issues with the ordering and can spend hours if not a day on this.
